src/plugins/intel_npu/src/plugin/npuw/partitioning/online/snapshot.hpp

```diff
@@ -89,6 +91,7 @@ class Snapshot : public std::enable_shared_from_this<Snapshot> {
     size_t getNextRepId();
 
     std::shared_ptr<ov::Model> m_model;
+    ::intel_npu::Config& m_cfg;
```

Comment:
As an entity, snapshot shouldn't be exposed to the plugin configuration - as it opens new possibilities (for the abstraction leaks). Instead, let's just detect this as a feature and then, looking at that feature, disable the optimization we know can be harmful